edge-modules/TwinTester/Settings.cs

```diff
@@ -20,26 +20,32 @@ public class Settings
                     .AddEnvironmentVariables()
                     .Build();
 
-                string moduleId = configuration.GetValue<string>("TargetModuleId");
-                moduleId = string.IsNullOrWhiteSpace(moduleId) ? configuration.GetValue<string>("IOTEDGE_MODULEID", string.Empty) : moduleId;
+                string moduleId = configuration.GetValue<string>("IOTEDGE_MODULEID", string.Empty);
+                string targetModuleId = configuration.GetValue<string>("TargetModuleId", moduleId);
+
+                string analyzerUrl = configuration.GetValue<string>("AnalyzerUrl", "http://analyzer:15000");
+                string reporterUrl = configuration.GetValue<string>("testResultCoordinatorUrl", analyzerUrl);
```

Comment:
Why do we need both of these endpoints? Your later changes only include reporterUrl, which I thought would be a general reporter (configurable for wherever we report to)

Reply:
Because TwinTester reports to analyzer in long haul and testResultCoordinator in connectvity tests. So reporterUrl will try to use testResultCoordinatorUrl, if it doesn't exist use analyzerUrl

Reply:
since either testResultCoordinatorUrl or analyzerUrl will be used and assigned to reportUrl; should we just define a report url in config file?

Reply:
Another point is, we should handle this in a consistent way in both TwinTester and DirectMethodSender projects.  Please sync up with @yophilav.

Reply:
As discussed, refactoring PR for direct method should take this into consideration

---
In reply to: [361517522](https://github.com/Azure/iotedge/pull/2177#discussion_r361517522) [](ancestors = 361517522)